Backend/Controllers/ProductController.cs

```diff
@@ -0,0 +1,250 @@
+                if (shop == null) return Forbid();
+
+                if (product == null) return NotFound();
+
+
+                if (IsImageFile(imageFile))
+                {
+                     var fileId = Guid.NewGuid().ToString() + Path.GetExtension(imageFile.FileName);
+
+                     using (var stream = imageFile.OpenReadStream())
+                     {
+                        var uploadedFile = await storage.Upload(fileId, stream);
+
+                        if (uploadedFile != null)
+                        {
+                            var productImage = new ProductImage(uploadedFile.Provider, uploadedFile.Bucket, uploadedFile.Key, id);
+                            db.ProductImages.Add(productImage);
+                        }
+                     }
+                    var saved = await db.Save();
+                    return saved ? Ok() : Problem();
+                }
+
+            return Problem();
+         
+        }
+
+
+        public static bool IsImageFile(IFormFile file)
+        {
+            if (file == null || string.IsNullOrEmpty(file.FileName) || file.Length == 0)
+            {
+                return false;
+            }
+
+            var fileExtension = Path.GetExtension(file.FileName).ToLower();
+            string[] photoExtensions = { ".jpg", ".jpeg", ".png", ".gif", ".bmp" };
+
+            return photoExtensions.Contains(fileExtension);
+        }
+
+        [HttpDelete("{id}/image")]
+        public async Task<IActionResult> DeleteProductImage(string fileId)
+        {
+            var productImagesToDelete = await db.ProductImages.QueryOne(pi => pi.Id == fileId);
+
+            var uid = User.FindFirst(Jwt.Uid)?.Value;
+            var shop = await db.Shops.QueryOne(s => s.Id == productImagesToDelete.Product.ShopId && s.OwnerId == uid);
+
+            if (shop == null) return Forbid();
+
+            bool isDeleted = await storage.Delete(productImagesToDelete);
+
+            if (isDeleted)
+            {
+                var saved = await db.Save();
+                return saved ? Ok() : Problem();
+            }
+
+            return Problem();
+         
+        }
+
+
+        [HttpDelete("{id}")]
+        public async Task<IActionResult> DeleteProduct(string id)
+        {
+            var product = await db.Products.QueryOne(p => p.Id == id);
+
+            var uid = User.FindFirst(Jwt.Uid)?.Value;
+            var shop = await db.Shops.QueryOne(s => s.Id == product.ShopId && s.OwnerId == uid);
+
+            if (shop == null) return Forbid();
+
+            if (product == null) return NotFound();
+
+            bool canDeleteProduct = !await db.Orders.AnyAsync(o => o.ProductId == id);
+
+            if (canDeleteProduct)
+            {
+                var images = await db.ProductImages.QueryMany(x => x.ProductId == product.Id);
+                await imageService.SafeDelete(images);
+                db.ProductImages.RemoveRange(images);
+                db.Products.Remove(product);
```

Comment:
no need for `db.ProductImages.RemoveRange(images);` in 240 line. Because images connected to product in database, so when product is deleted it will delete all images